plugins/nvm/nvm.plugin.zsh

```diff
@@ -16,14 +16,37 @@ fi
 # Note: nvm is a function so we need to use `which`
 which nvm &>/dev/null && return
 
+# TODO: 2022-11-11: Remove soft-deprecate options
+if (( $+NVM_LAZY + $+NVM_LAZY_CMD + $+NVM_AUTOLOAD)); then
+  echo "[oh-my-zsh] We moved nvm plugin configuration to zstyle"
+  echo "[oh-my-zsh] Get more information in https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/nvm"
+fi
 if (( $+NVM_LAZY )); then
-  # Call nvm when first using nvm, node, npm, pnpm, yarn or $NVM_LAZY_CMD
-  function nvm node npm pnpm yarn $NVM_LAZY_CMD {
-    unfunction nvm node npm pnpm yarn $NVM_LAZY_CMD
+  echo "[oh-my-zsh] Instead of NVM_LAZY=1, please use"
+  echo "[oh-my-zsh] zstyle ':omz:plugins:nvm' lazy true"
+  zstyle ':omz:plugins:nvm' lazy true
+fi
+if (( $+NVM_LAZY_CMD )); then
+  echo "[oh-my-zsh] Instead of NVM_LAZY_CMD=$NVM_LAZY_CMD, please use"
+  echo "[oh-my-zsh] zstyle ':omz:plugins:nvm' lazy-cmd $NVM_LAZY_CMD"
+  zstyle ':omz:plugins:nvm' lazy-cmd $NVM_LAZY_CMD
+fi
+if (( $+NVM_AUTOLOAD )); then
+  echo "[oh-my-zsh] Instead of NVM_LAZY=1, please use"
+  echo "[oh-my-zsh] zstyle ':omz:plugins:nvm' lazy true"
+  zstyle ':omz:plugins:nvm' autoload true
+fi
```

Comment:
Not a fan of these being so verbose, I think it's going to scare some people and will not read the entire text. I think it's better to do it once, write the output to stderr, and separate the information text and the alternative `zstyle` commands, sort of like this:

```
# text in yellow
[nvm plugin] Variable-style settings are deprecated, instead of $<variable-names-used-by-the-user>, use:

[nvm plugin] zstyle ':omz:plugins:nvm' lazy
[nvm plugin] zstyle ':omz:plugins:nvm' lazy-cmd
[nvm plugin] zstyle ':omz:plugins:nvm' autoload
# ^ this list only shows the settings used by the user
```

Also good call in writing the exact `lazy-cmd` value used by the user in the warning.